### DISCUSSION TITLE
Could not find the sourceStructure.json file

### DISCUSSION THREAD

**Discussion by aso-toshihiko** [ExploreASL/ExploreASL#1250] [Q&A]
Hi,

After trying the GUI and found it very useful to create the json files, I'm now trying the beta version (ExploreASL-develop is the beta version, am I right?) from Matlab R2020 to see how it works without GUI. This was suggested in this discussion forum regarding the Structure error related to CAT12.

The ExploreASL-dev was downloaded in November.

I prepared the dicom file which worked perfectly fine with the GUI:
![キャプチャ](https://user-images.githubusercontent.com/55236088/205006561-0833355c-e4ae-44ed-94a7-4a4a2f699813.JPG)

Edited the json files to fit this single subject longitudinal data.
And evoked ExploreASL from its directory, which gave me the error. What should I check?:

>> [x] = ExploreASL(' D:\Saito\kaiseki_s04', [1 1 0], 1);

Neither the sourceStructure.json, dataset_description.json nor dataPar.json exist, ExploreASL will only be initialized...
ExploreASL will run the import workflow and will load the dataset...
==============================================================================================
 ________                      __                                 ______    ______   __          
/        |                    /  |                               /      \  /      \ /  |       
########/  __    __   ______  ## |  ______    ______    ______  /######  |/######  |## |         
##    |   ##  \/##/ /######  |## |/######  |/######  |/######  |##    ## |##      \ ## |       
#####/     ##  ##<  ## |  ## |## |## |  ## |## |  ##/ ##    ## |######## | ######  |## |         
##       |/##/ ##  |##    ##/ ## |##    ##/ ## |      ##       |## |  ## |##    ##/ ##       |   
########/ ##/   ##/ #######/  ##/  ######/  ##/        #######/ ##/   ##/  ######/  ########/    
                    ## |                                                                         
                    ## |                                                                         
                    ##/                                                                          

======================================= ExploreASL Settings ==================================
Dataset Root         D:\Saito\kaiseki_s04
Import & Defacing   DCM2NII NII2BIDS 
Processing          
bPause              False
iWorker             1
nWorkers            1
==============================================================================================
ExploreASL v1.10.0 (beta) initialized... 
Click here for the ExploreASL manual

======================================== ExploreASL Import ===================================
Click here for the ExploreASL DICOM to BIDS (import) tutorial
ExploreASL Import initialization failed...
Could not find the sourceStructure.json file...
xASL_imp_BasicParameterChecks, line 42...
Warning: Loading subjects didnt work, verify that ExploreASL was correctly initialized with
the correct DataParameter file...
Reference to non-existent field 'SUBJECTS'. 
> In xASL_init_Iteration (line 69)
  In ExploreASL_Import (line 50)
  In ExploreASL (line 91) 

Running xASL_module_Import...
Reference to non-existent field 'SESSIONS'.

Error in xASL_init_Iteration (line 108)
        dbSettings.sets.SESSION = x.SESSIONS;

Error in ExploreASL_Import (line 50)
	[~, x] = xASL_init_Iteration(x,'xASL_module_Import');

Error in ExploreASL (line 91)
        ExploreASL_Import(x);

**Comment by MDijsselhof** [comment]
Hi,

You are correct, the ExploreASL-beta is the version of the develop branch. Regarding your errors: it may be due to an incorrect location of the files mentioned in the error. Can you specify where the files are located? They should be in ` D:\Saito\kaiseki_s04`, however it appears they are located in ` D:\Saito\kaiseki_s04\sourcedata`. 

Kind regards,

Mathijs

**Reply by aso-toshihiko** [reply]
Hi, thanks for the response!

I have never put any .json files in the "sourcedata" directory. What made you think it's located in that directory?
All .json files are directly under the D:\Saito\kaiseki_s04 directory. Strange!

**Reply by MDijsselhof** [reply]
Hi,

The image you posted made it appear like the .json locations was wrong, but indeed then this should have been fine! Where you call ExploreASL, in `[x] = ExploreASL(' D:\Saito\kaiseki_s04', [1 1 0], 1);` can you check if there is a space in front of the dataset path? To be sure this is the issue, you could try `[x] = ExploreASL('D:\Saito\kaiseki_s04', [1 1 0], 1);`. Please let us know if this fixes the errors!

Kind regards,

Mathijs
